docs/core/deploying/ready-to-run.md

```diff
@@ -0,0 +1,65 @@
+
+R2R binaries improve startup performance by reducing the amount of work the just-in-time (JIT) compiler needs to do as your application loads. The binaries contain similar native code compared to what the JIT would produce. However, R2R binaries are larger because they contain both intermediate language (IL) code, which is still needed for some scenarios, and the native version of the same code. R2R is only available when you publish a self-contained app that targets specific runtime environments (RID) such as Linux x64 or Windows x64.
+
+To compile your project as ReadyToRun, do the following:
+
+01. Add the `<PublishReadyToRun>` setting to your project:
+
+    ```xml
+    <PropertyGroup>
+      <PublishReadyToRun>true</PublishReadyToRun>
+    </PropertyGroup>
+    ```
+
+02. Publish an application. For example, this command creates a app for the 64-bit version of Windows:
+
+    ```dotnetcli
+    dotnet publish -c Release -r win-x64
+
+    ```
+
+## Impact of using the ReadyToRun feature
+
+### General impact
+Ahead of time compilation has somewhat complex and unpredictable performance impacts on application performance. In general the size of an assembly will grow to between 2 to 3 times larger. This increase in the physical size of the file may reduce the performance of loading the assembly from disk, and increase working set of the process. However, in return the number of methods compiled is typically reduced substantially. The result is that most applications that consist of a substantial amount of code receive large performance benefits from enabling ReadyToRun. Applications which have small amounts of code will likely not experience a significant improvement from enabling ReadyToRun, as the .NET framework assemblies have already been precompiled with ReadyToRun.
+
+Note that the startup improvement discussed here applies not only to application startup, but also to the first use of any code in the application. For instance, ReadyToRun can be used to reduce the response latency of the first use  of a WebAPI in an ASP.NET application.
+
+### Interaction with tiered compilation
+Ahead of time code generation quality is generally slightly worse than the quality of code produced by the JIT. To address this issue, tiered compilation will replace commonly used ReadyToRun methods with JIT generated methods.
+
+## How is the set of precompiled assemblies chosen?
+The SDK will precompile the assemblies that are distributed with the application. For self-contained applications this will include recompiling the framework. C++/CLI binaries are not eligible for ReadyToRun compilation.
+
+## How is the set of methods to precompile chosen?
+The compiler will attempt to pre-compile as many methods as it can. However various reasons it is not expected that using the ReadyToRun feature will result in preventing the JIT from executing.
```

Comment:
Nit - should this be "However *for* various reasons ..."?

Reply:
> preventing the JIT from executing

I would rather word it like this:
> The compiler will attempt to pre-compile as many methods as it can. However, for various reasons some methods may still need to be compiled by the JIT at run time.